tests/infrastructure/tekton/utils.py

```diff
@@ -94,3 +106,28 @@ def process_yaml_files(file_paths, replacements, resource_kind, namespace):
         update_tekton_resources_yaml_file(file_path=file_path, replacements=replacements)
         resources.append(resource_kind(yaml_file=file_path, namespace=namespace).create())
     return resources
+
+
+def wait_and_assert_pipelinerun_succeeded(
+    pipelinerun: PipelineRun, wait_timeout: int = TIMEOUT_50MIN, sleep_interval: int = TIMEOUT_1MIN
+) -> None:
+    try:
+        for sample in TimeoutSampler(
+            wait_timeout=wait_timeout,
+            sleep=sleep_interval,
+            func=lambda: pipelinerun.instance.status.conditions[0],
+        ):
+            if sample.status != Resource.Condition.Status.UNKNOWN:
+                LOGGER.info(f"PipelineRun Condition : {sample}")
+                break
+
+        assert sample.status == Resource.Condition.Status.TRUE and sample.type == Resource.Condition.Phase.SUCCEEDED, (
+            f"Pipelines failed to succeed. Pipeline status: {pipelinerun.instance.status}"
+        )
+
+    except TimeoutExpiredError:
+        LOGGER.error(
+            f"Pipelinerun: {pipelinerun.name}, "
+            f"Preparing for VM teardown due to TimeoutError. Last available conditions: {sample}"
```

Comment:
I don't understand this error message. This function wait_and_assert_pipelinerun_succeeded() is generic. Where is this vm teardown coming from?

Reply:
I will improve error message. It's the old one from previous code and can be more generic.